From bec6b6b117a5a2bca1b1ee539ece09eca2471eb3 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Mon, 2 Nov 2020 15:23:29 -0600
Subject: [PATCH] TF-A - fix PFC related initialization regarding Beacon
 EmbeddedWorks kit(s)

Don't configure GPIOs as outputs that are expected to be inputs
Set POCCTRL0 correctly for SD1 signals

Signed-off-by: Charles Stevens <charles.stevens@compasses.com>
---
 plat/renesas/rcar/pfc/H3/pfc_init_h3_v2.c | 16 +++++++++++-----
 plat/renesas/rcar/pfc/M3/pfc_init_m3.c    | 14 ++++++++++----
 plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c  | 16 +++++++++++-----
 3 files changed, 32 insertions(+), 14 deletions(-)
 mode change 100644 => 100755 plat/renesas/rcar/pfc/H3/pfc_init_h3_v2.c
 mode change 100644 => 100755 plat/renesas/rcar/pfc/M3/pfc_init_m3.c

diff --git a/plat/renesas/rcar/pfc/H3/pfc_init_h3_v2.c b/plat/renesas/rcar/pfc/H3/pfc_init_h3_v2.c
old mode 100644
new mode 100755
index 91137dea..66d29e69
--- a/plat/renesas/rcar/pfc/H3/pfc_init_h3_v2.c
+++ b/plat/renesas/rcar/pfc/H3/pfc_init_h3_v2.c
@@ -1436,7 +1436,13 @@ void pfc_init_h3_v2(void)
 
 	/* initialize POC control register */
 #if RZG_HIHOPE_RZG2H
-	pfc_reg_write(PFC_POCCTRL0, POC_SD0_DAT3_33V
+	pfc_reg_write(PFC_POCCTRL0, POC_SD1_DAT3_33V
+				  | POC_SD1_DAT2_33V
+				  | POC_SD1_DAT1_33V
+				  | POC_SD1_DAT0_33V
+				  | POC_SD1_CMD_33V
+				  | POC_SD1_CLK_33V
+				  | POC_SD0_DAT3_33V
 				  | POC_SD0_DAT2_33V
 				  | POC_SD0_DAT1_33V
 				  | POC_SD0_DAT0_33V
@@ -2038,10 +2044,10 @@ void pfc_init_h3_v2(void)
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
 	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
 	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
-	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
-	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
-	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00000F00U);
 	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
 #else  // RZG_HIHOPE_RZG2H
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000000U);
diff --git a/plat/renesas/rcar/pfc/M3/pfc_init_m3.c b/plat/renesas/rcar/pfc/M3/pfc_init_m3.c
old mode 100644
new mode 100755
index a5076afe..75167589
--- a/plat/renesas/rcar/pfc/M3/pfc_init_m3.c
+++ b/plat/renesas/rcar/pfc/M3/pfc_init_m3.c
@@ -1517,7 +1517,13 @@ void pfc_init_m3(void)
 
 	/* initialize POC control register */
 #if RZG_HIHOPE_RZG2M
-	pfc_reg_write(PFC_POCCTRL0, POC_SD0_DAT3_33V
+	pfc_reg_write(PFC_POCCTRL0, POC_SD1_DAT3_33V
+				  | POC_SD1_DAT2_33V
+				  | POC_SD1_DAT1_33V
+				  | POC_SD1_DAT0_33V
+				  | POC_SD1_CMD_33V
+				  | POC_SD1_CLK_33V
+				  | POC_SD0_DAT3_33V
 				  | POC_SD0_DAT2_33V
 				  | POC_SD0_DAT1_33V
 				  | POC_SD0_DAT0_33V
@@ -2119,10 +2125,10 @@ void pfc_init_m3(void)
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
 	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
 	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
-	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
-	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
 	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
-	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00000F00U);
 	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
 #else  // RZG_HIHOPE_RZG2M
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000000U);
diff --git a/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c b/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c
index fc1cdb2a..fbccf4ce 100644
--- a/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c
+++ b/plat/renesas/rcar/pfc/M3N/pfc_init_m3n.c
@@ -1424,7 +1424,13 @@ void pfc_init_m3n(void)
 
 	/* initialize POC control register */
 #if RZG_HIHOPE_RZG2N
-	pfc_reg_write(PFC_POCCTRL0, POC_SD0_DAT3_33V
+	pfc_reg_write(PFC_POCCTRL0, POC_SD1_DAT3_33V
+				  | POC_SD1_DAT2_33V
+				  | POC_SD1_DAT1_33V
+				  | POC_SD1_DAT0_33V
+				  | POC_SD1_CMD_33V
+				  | POC_SD1_CLK_33V
+				  | POC_SD0_DAT3_33V
 				  | POC_SD0_DAT2_33V
 				  | POC_SD0_DAT1_33V
 				  | POC_SD0_DAT0_33V
@@ -2026,10 +2032,10 @@ void pfc_init_m3n(void)
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
 	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
 	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
-	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
-	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
-	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00000F00U);
 	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
 #else  // RZG_HIHOPE_RZG2N
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000000U);
-- 
2.17.1

